### ISSUE TITLE
Question Regarding Target Rule Related and Upsell products

### ISSUE THREAD

**Issue by aaronware** [Smile-SA/elasticsuite#445]
Wondering if this module affect the related, upsell and cross sell for target rules within EE 

### Preconditions
PHP 7.0.X
Magento 2.1.7
Nginx
Varnish

Magento Version : 2.1.7 EE
ElasticSuite Version : 2.3.0
Environment : Developer

Third party modules : Some home grown, some Amasty

### Steps to reproduce
1. Create Upsell Rule proper match and result
2. Pick products based on "collection" attribute
3. Zero Results after reindex.

### Expected result
1. Show new products

### Actual result
Nothing displays

**Comment by romainruaud** [comment]
Hello @aaronware 

ElasticSuite does not affect the TargetRule or, at least, should not. We did not face any issues on TargetRules during our tests.

But TargetRule are known to be buggy on EE.

Also, they are also considered as a perf killer regarding the way they work by using huge "JOIN" statements across many DB tables.

That's why we did a custom module to build their conditions with ElasticSuite. The module can be found here : https://github.com/Smile-SA/magento2-module-elasticsuite-targetrule

Give it a try and let us know if it's better.

Regards
